### PR TITLE
feat: Add support for kvv2

### PR DESCRIPTION
**Pull request by werne2j**
### Description
These changes allow for the user to specify which kv version to be used. This will now allow us to support kv1 and kv2.

**Fixes:** #42 

### Checklist
Please make sure that your PR fulfills the following requirements: 
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been
- [x] Docs have been added / updated

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

**Comment by jkayani** [comment]
I tested this build:

```
default@argocd-repo-server-7467f69495-f8rcr:/tmp/some-path/infrastructure/temp-test$ argocd-vault-plugin generate -s local-vault .
apiVersion: v1
data:
  test: eyAiZm9vIjogMTIzLCAiYmFyIjogNDU2IH0=
kind: Secret
metadata:
  annotations:
    avp_path: secret/data/regsecret
  name: regsecret-test
  namespace: argocd
type: Opaque
---
default@argocd-repo-server-7467f69495-f8rcr:/tmp/some-path/infrastructure/temp-test$ exit


Joshuas-MBP-2:ops jkayani$ vault kv get secret/regsecret
====== Metadata ======
Key              Value
---              -----
created_time     2021-02-04T19:19:18.895515Z
deletion_time    n/a
destroyed        false
version          1

======= Data =======
Key            Value
---            -----
pull-secret    { "foo": 123, "bar": 456 }
Joshuas-MBP-2:ops jkayani$
```

It worked from the UI too. I just did it from the console to easily demonstrate

**Comment by jkayani** [comment]
Screen Shot 2021-02-04 at 14 45 57

Also confirm that specifying the engine-version only as annotation worked

**Comment by cuttingedge1109** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -50,12 +50,21 @@ The plugin requires some configuration to connect to Vault. The parameters are:
...
 | SECRET_ID      | Vault AppRole Secret_ID    | Required with `AUTH_TYPE` of `approle` |
 | IBM_API_KEY    | IBM Cloud IAM API Key      | Required with `TYPE` of `secretmanager` and `AUTH_TYPE` of `iam` |
 
+#### Supported Annotation
+We support two different annotations that can be used inside a kubernetes resource. These annotations will override any corrisponding configuration set via Environment Variable or Configuration File.
+
+| Annotation | Description |  
+| ---------- | ----------- |  
+| avp_path | Path to the Vault Secret |
+| kvVersion | Version of the KV Secret Engine |
```
Comment (jkayani) [inline README.md:66]: I'd almost say we should decide on under_case or camelCase and be consistent, but idk if that really matters

**Review by jkayani — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (werne2j) [reply README.md:66]: Hmm good call

**Review by jkayani — APPROVED** [review]

**Pull request merged**